consensus/src/alerts/service.rs

```diff
@@ -0,0 +1,148 @@
+use crate::{
+    alerts::{
+        handler::Handler, io::IO, Alert, AlertConfig, AlertMessage, AlerterResponse,
+        ForkingNotification, NetworkMessage,
+    },
+    Data, Hasher, MultiKeychain, Multisigned, Receiver, Recipient, Sender, Terminator,
+};
+use aleph_bft_rmc::{DoublingDelayScheduler, Message as RmcMessage, ReliableMulticast};
+use futures::{channel::mpsc, FutureExt, StreamExt};
+use log::{debug, error, warn};
+use std::time;
+
+pub struct Service<H: Hasher, D: Data, MK: MultiKeychain> {
+    handler: Handler<H, D, MK>,
+    io: IO<H, D, MK>,
+}
+
+impl<H: Hasher, D: Data, MK: MultiKeychain> Service<H, D, MK> {
+    pub fn new(
```

Comment:
I know you want to do the simulations in a separate PR (rightly so!), but this is still weird. I would suggest more something like:
1. Merge `Service` and `IO` (probably using the former name) to contain all the IO, but no `Handler`.
2. Have a `run` method that takes the `Handler` as an argument.
3. Create both in `member` and (for now) just pass one to the other.

This should make it simpler to do the simulation in the next PR, but if done properly should also remove the need for `alerter_index` (since you'll have access to the `Handler`), as well as remove the `pub`s in `IO`s fields. In the future if we also refactor `Rmc` it would also likely be simpler to slot it in here.

Reply:
Oh, and `exiting` shouldn't need to be passed as an argument (although `Handler` might have to sometimes).

Reply:
how about passing `Handler` to `Service::new`?